lib/app/modules/settings/controllers/theme_controller.dart

```diff
@@ -1,32 +1,99 @@
+import 'dart:async';
 import 'package:flutter/material.dart';
 import 'package:get/get.dart';
 import 'package:ultimate_alarm_clock/app/data/providers/secure_storage_provider.dart';
 import 'package:ultimate_alarm_clock/app/utils/constants.dart';
+import 'dart:developer' as dev;
 
 class ThemeController extends GetxController {
   var isLightMode = true.obs;
   final _secureStorageProvider = SecureStorageProvider();
 
+  late Timer _brightnessCheckTimer;
+  bool isSystemModeActive = false;
+
   @override
   void onInit() {
     _loadThemeValue();
     super.onInit();
   }
 
+  @override
+  void onClose() {
+    // _brightnessCheckTimer.cancel();
+    super.onClose();
+  }
+
   void _loadThemeValue() async {
-    isLightMode.value =
-        await _secureStorageProvider.readThemeValue() == AppTheme.light;
-    Get.changeThemeMode(isLightMode.value ? ThemeMode.light : ThemeMode.dark);
+    final storedTheme = await _secureStorageProvider.readThemeValue();
+    dev.log(storedTheme.toString());
+    if (storedTheme == AppTheme.system) {
+      _startBrightnessCheckTimer();
+      final systemBrightness = Get.mediaQuery.platformBrightness;
+      isSystemModeActive = true;
+      if (systemBrightness == Brightness.light) {
+        isLightMode.value = true;
+        Get.changeThemeMode(ThemeMode.light);
+        onClose();
+      } else {
+        isLightMode.value = false;
+        Get.changeThemeMode(ThemeMode.dark);
+        onClose();
+      }
+    } else {
+      onClose();
+      isLightMode.value = storedTheme == AppTheme.light;
+      Get.changeThemeMode(isLightMode.value ? ThemeMode.light : ThemeMode.dark);
+    }
   }
 
-  void _saveThemeValuePreference() async {
-    await _secureStorageProvider.writeThemeValue(
-      theme: isLightMode.value ? AppTheme.light : AppTheme.dark,
-    );
+  void _startBrightnessCheckTimer() {
+    _brightnessCheckTimer = Timer.periodic(Duration(seconds: 2), (timer) {
```

Comment:
This Timer is going to be resource intensive for something as simple as a theme change. We can just make it work on app startup and not keep it "real time"

Reply:
But if so then it wouln't check for light or dark difference right?

Reply:
Wait, as I understand this timer is to check if the the device's theme is dark or light and it updates the theme on the app right? The timer runs every 2 seconds to check this, right?

I think we can just set the app to the appropriate theme when it's opened up rather than having to change it in realtime.

Reply:
Correct me if I am wrong, but isn't that how system theme works? As in if my app in System mode, it needs to check if my device is in light or dark mode right? It is always listening for changes, If we are checking just at the start of the app then its going to be simply like a default light and dark mode yea?

Reply:
Ok. Fine. I will make that change.